subsys/modem/modem_chat.c

```diff
@@ -0,0 +1,789 @@
+		if ((chat->parse_match->separators[i]) ==
+		    (chat->receive_buf[chat->receive_buf_len - 1])) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool modem_chat_parse_end_del_start(struct modem_chat *chat)
+{
+	for (uint8_t i = 0; i < chat->delimiter_size; i++) {
+		if (chat->receive_buf[chat->receive_buf_len - 1] == chat->delimiter[i]) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool modem_chat_parse_end_del_complete(struct modem_chat *chat)
+{
+	/* Validate length of end delimiter */
+	if (chat->receive_buf_len < chat->delimiter_size) {
+		return false;
+	}
+
+	/* Compare end delimiter with receive buffer content */
+	return (memcmp(&chat->receive_buf[chat->receive_buf_len - chat->delimiter_size],
+		       chat->delimiter, chat->delimiter_size) == 0)
+		       ? true
+		       : false;
+}
+
+static void modem_chat_on_command_received_unsol(struct modem_chat *chat)
+{
+	/* Callback */
+	if (chat->parse_match->callback != NULL) {
+		chat->parse_match->callback(chat, (char **)chat->argv, chat->argc, chat->user_data);
+	}
+}
+
+static void modem_chat_on_command_received_abort(struct modem_chat *chat)
+{
+	/* Callback */
+	if (chat->parse_match->callback != NULL) {
+		chat->parse_match->callback(chat, (char **)chat->argv, chat->argc, chat->user_data);
+	}
+
+	/* Abort script */
+	modem_chat_script_stop(chat, MODEM_CHAT_SCRIPT_RESULT_ABORT);
+}
+
+static void modem_chat_on_command_received_resp(struct modem_chat *chat)
+{
+	/* Callback */
+	if (chat->parse_match->callback != NULL) {
+		chat->parse_match->callback(chat, (char **)chat->argv, chat->argc, chat->user_data);
+	}
+
+	/* Advance script */
+	modem_chat_script_next(chat, false);
+}
+
+static bool modem_chat_parse_find_catch_all_match(struct modem_chat *chat)
+{
+	/* Find in all matches types */
+	for (uint16_t i = 0; i < ARRAY_SIZE(chat->matches); i++) {
+		/* Find in all matches of matches type */
+		for (uint16_t u = 0; u < chat->matches_size[i]; u++) {
+			/* Validate match config is matching previous bytes */
+			if (chat->matches[i][u].match_size == 0) {
+				chat->parse_match = &chat->matches[i][u];
+				chat->parse_match_type = i;
+
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+static void modem_chat_on_command_received(struct modem_chat *chat)
+{
+	LOG_DBG("\"%s\"", chat->argv[0]);
+
+	switch (chat->parse_match_type) {
+	case MODEM_CHAT_MATCHES_INDEX_UNSOL:
+		modem_chat_on_command_received_unsol(chat);
+		break;
+
+	case MODEM_CHAT_MATCHES_INDEX_ABORT:
+		modem_chat_on_command_received_abort(chat);
+		break;
+
+	case MODEM_CHAT_MATCHES_INDEX_RESPONSE:
+		modem_chat_on_command_received_resp(chat);
+		break;
+	}
+}
+
+static void modem_chat_on_unknown_command_received(struct modem_chat *chat)
+{
+	/* Try to find catch all match */
+	if (modem_chat_parse_find_catch_all_match(chat) == false) {
+		return;
+	}
+
+	/* Terminate received command */
+	chat->receive_buf[chat->receive_buf_len - chat->delimiter_size] = '\0';
+
+	/* Parse command */
```

Comment:
A `LOG_DBG` to print any unhandled command would be great.
I like the option to see "all" unhandled command traffic.

Reply:
Added